### ISSUE TITLE
Variables: Save dashboard variables per user

### ISSUE THREAD

**Issue by gelicia** [grafana/grafana#82999]
### Discussed in https://github.com/grafana/grafana/discussions/44523

<div type='discussions-op-text'>

<sup>Originally posted by **yoyo421** January 27, 2022</sup>
**What would you like to be added:**
Add a fallback if a dashboard variable is not set in the URL query, check what the user set the variable last time

**Why is this needed:**
If I have a couple of variables, that uses a lot of options, every time I open the dashboard I need to start selecting the variables again from a very long list of options, and it gets really frustrating really fast, because I'm not the owner of the dashboard I cannot save my changes and keep it for the next time (other people working on this dashboard but interested in a different set of selections).

Keeping track of the URL every time I change a variable will not help because I open the dashboard from Grafana where the default URL is without the variables.</div>

**Comment by pathakhimanshucs** [comment]
Agreed, we have lots of filters and it can be useful sometimes to start again with the previous filters. In this case a reset button would be useful that returns all variables to the dashboard defaults.

**Comment by shousper** [comment]
Even as an opt-in feature this would be great for larger teams.

Many "universal" dashboards like looking at general resource consumption for containers, processes, namespaces, etc. have completely different use cases for different teams and users. Each interested in different instances, namespaces, and maybe even data sources.

Duplicating dashboards isn't particularly efficient here, and requires (ideally) a provisioning pipeline to configure variable defaults per team, and deploy copies per team. Or less ideally, teams duplicating dashboards and having to sync updates themselves or drift. Still, this totally obviates user-specific preferences, forcing users to save copies.

At the moment, I think everyone is either selecting variables each time they open a dashboard, or saving bookmarks somewhere (either their browser, or something like Notion). While these can sync between devices, I would value an native solution here that requires less clicking around.
